### PR TITLE
Updated from_repo_line in apt to generate valid list files

### PR DESCRIPTION
**Pull request by MiaAltieri**
Changed from_repo_line to convert '/' to '-' in release specifications
when generating list file
Added an additional test to verify this functionality

### CODE REVIEW

File: tests/integration/test_apt.py [hunk 1]
```diff
@@ -56,3 +56,23 @@ def test_install_package_external_repository():
...
+        line = "deb [ arch=amd64,arm64 ] https://repo.mongodb.org/apt/ubuntu focal/mongodb-org/5.0 multiverse"
+        repo = apt.DebianRepository.from_repo_line(line)
+        # Import the repository's key
+        repo.import_key(key)
+        repositories.add(repo)
+
+    apt.update()
+    apt.add_package("mongodb-org")
+
+    assert get_command_path("mongodb-org") == "/usr/bin/mongodb-org"
```
Comment (jnsgruk) [inline tests/integration/test_apt.py:78]: Something not quite right about this test -- it's failing in the CI. Should this not be `get_command_path("mongod")`? Needs looking into

File: lib/charms/operator_libs_linux/v0/apt.py [hunk 2]
```diff
@@ -206,13 +206,9 @@ def __eq__(self, other) -> bool:
...
-        return (
-            isinstance(other, self.__class__)
-            and (
-                self._name,
-                self._version.number,
-            )
-            == (other._name, other._version.number)
+        return isinstance(other, self.__class__) and (self._name, self._version.number) == (
+            other._name,
+            other._version.number,
```
Comment (jnsgruk) [inline lib/charms/operator_libs_linux/v0/apt.py:211]: Most of the changes in this PR seem to be formatting changes -- one to be aware of in future as it makes seeing the actual changes a little harder.

File: lib/charms/operator_libs_linux/v0/apt.py [hunk 3]
```diff
@@ -934,7 +918,9 @@ def from_repo_line(repo_line: str, write_file: Optional[bool] = True) -> "Debian
             write_file: boolean to enable writing the new repo to disk
         """
         repo = RepositoryMapping._parse(repo_line, "UserInput")
-        fname = f"{DebianRepository.prefix_from_uri(repo.uri)}-{repo.release}.list"
+        fname = (
+            f"{DebianRepository.prefix_from_uri(repo.uri)}-{repo.release.replace('/', '-')}.list"
```
Comment (jnsgruk) [inline lib/charms/operator_libs_linux/v0/apt.py:922]: I think this is right -- but I think a similar change is needed at line 1241?

**Review by jnsgruk — CHANGES_REQUESTED** [review]
Thanks @MiaAltieri!

A couple of changes please -- would be good to have a unit test in here too that tests the name of the created repo list files with an example where it's just a plain repo like `focal` vs. one like in the test case which is `focal/something/something`

File: lib/charms/operator_libs_linux/v0/apt.py (hunk 2 above)
Comment (MiaAltieri) [reply lib/charms/operator_libs_linux/v0/apt.py:211]: Ah I see, my bad, I ran black on this and it reformatted the file. Should I revert these changes?

Comment (jnsgruk) [reply lib/charms/operator_libs_linux/v0/apt.py:211]: Probably best just keep this PR clean. The code should have already been formatted through `tox -e lint` (which uses `black` with a line length of 99)

**Review by jnsgruk — APPROVED** [review]
Nice, thanks! One final change that would be useful in this PR is to increment the LIBPATCH by 1 so it's ready to publish once merged 🙌🏻

**Comment by jnsgruk** [comment]
@rbarry82 once the LIBPATCH is updated, can you merge this and publish the latest version of Charmhub please? (I'm afk today)

**Comment by rbarry82** [comment]
Definitely

**Comment by jnsgruk** [comment]
Merged and published, thanks!

**Pull request merged**